### PR TITLE
add new variables to make db more configureable

### PR DESCRIPTION
**Pull request by wodka**
Attempt for #23 - This would introduce several new configuration options:

```bash
OTRS_DATABASE = otrs
OTRS_DB_USER = otrs
OTRS_DB_HOST = mariadb
```

edit: removed settings that are no longer in this pull request

### CODE REVIEW

**Comment by juanluisbaptiste** [comment]
What's the point on using the otrs user for the database setup ? absolutely none, the setup can be done with the root user as is done already, you only need the otrs user to connect to the database from the OTRS container.

I will not merge this PR as it is.

**Comment by wodka** [comment]
I can change it so the setup is done with the root user again - I missed the create_db call in the beginning - thought I could with this eliminate the requirement to have root access by otrs in any way.

would it be ok for you if I split the installation part in 2 steps:

1) to create the db if it does not exist with the root user
2) to setup the db with the otrs user if no tables are present

this way it would be possible to not require a root user

**Comment by juanluisbaptiste** [comment]
I don't understand why you want to avoid the root user. It is only used to create the database and load up the schemas, there after the otrs user is used. This is standard way of doing it.

**Comment by wodka** [comment]
I would like to not give the image root permission, if I create the db and the user before it should not be required. Also not to check if the db is already up.

But if you like I could only add the custom user and db host part

**Comment by juanluisbaptiste** [comment]
I think you are not understanding how the image works. You need BOTH users, first root to create the database AND the otrs user. Then the otrs application uses the otrs EXCLUSIVELY (it's hardcoded on the configuration), the root user is ever again used. Even with your changes the root user will still exist, so what's the point ?

The only change I can allow is the environment variables to configure a different db server and port. Don't forget to update the documentation too.

**Comment by wodka** [comment]
I would actually prefer to be able to do the setup without having the root user. Same goes for the "wait_for_db()" function that requires the root user :/ - could we change at least the wait_for_db function to also work without root permissions (perhaps something like https://github.com/vishnubob/wait-for-it)?

update the code right now, will add that to the documentation as well.

**Comment by juanluisbaptiste** [comment]
You need the root user on wait_for_db to check when it's up. The check it does althought can be changed so it doesn't check the uer table. For now I'm not going to implement the wait-for-it script. If you want continue working on the env vars, but no other change I will add for now (I'm also working on OTRS 6 update).

**Comment by wodka** [comment]
thanks :)

the pull request right now only contains the changes for the 3 vars (OTRS_DATABASE, OTRS_DB_USER, OTRS_DB_HOST) plus the readme.

I removed everything else I did

**Comment by juanluisbaptiste** [comment]
I will check it now. About the db up check, we can continue to review it, but in a separate issue, to not mix both issues.

File: README.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ There are also some other environment variables that can be set to customize
 the default install:
 
 * `OTRS_HOSTNAME` Sets the container's hostname (auto-generated if not defined).
+* `OTRS_DB_NAME` name of database to use. Default is `otrs`.
```
Comment (juanluisbaptiste) [inline README.md]: Please Capitalize The First letter in all added env vars.

File: README.md [hunk 2]
```diff
@@ -8,12 +8,12 @@ Unofficial [OTRS 6 Free](http://www.otrs.com/software/) docker image. This repos
 
 We also include a *MariaDB Dockerfile* for a pre-configured image with the [required database settings](http://otrs.github.io/doc/manual/admin/stable/en/html/installation.html).
 
-The OTRS image doesn't include a SMTP service, decoupling applications into multiple containers makes it
```
Comment (juanluisbaptiste) [inline README.md]: Please remove from commit all changes added by a white space at the end of some lines, this is an unnecessary change that just pollutes the feature commit.

File: otrs/functions.sh [hunk 3]
```diff
@@ -32,11 +32,13 @@ OTRS_BACKUP_DIR="/var/otrs/backups"
...
 OTRS_CONFIG_MOUNT_DIR="/Kernel"
-OTRS_DATABASE="otrs"
 
 [ -z "${OTRS_INSTALL}" ] && OTRS_INSTALL="no"
+[ -z "${OTRS_DB_NAME}" ] && OTRS_DB_NAME ="otrs"
+[ -z "${OTRS_DB_USER}" ] && OTRS_DB_USER ="otrs"
+[ -z "${OTRS_DB_HOST}" ] && OTRS_DB_HOST ="mariadb"
 
-mysqlcmd="mysql -uroot -h mariadb -p${MYSQL_ROOT_PASSWORD} "
+mysqlcmd="mysql -uroot -h ${OTRS_DB_HOST} -p${OTRS_DB_PASSWORD} "
```
Comment (juanluisbaptiste) [inline otrs/functions.sh]: Shouldn't it be here MYSQL_ROOT_PASSWORD instead of OTRS_DB_PASSWORD as it was before ?

File: otrs/functions.sh [hunk 4]
```diff
@@ -141,7 +144,7 @@ function add_config_value() {
 
 function set_variables() {
   [ -z "${OTRS_HOSTNAME}" ] && OTRS_HOSTNAME="otrs-`random_string`" && print_info "OTRS_HOSTNAME not set, setting hostname to '${OTRS_HOSTNAME}'"
-  [ -z "${OTRS_DB_PASSWORD}" ] && OTRS_DB_PASSWORD=`random_string` && print_info "OTRS_DB_PASSWORD not set, setting password to '${OTRS_DB_PASSWORD}'"
+  [ -z "${OTRS_DB_PASSWORD}" ] && OTRS_DB_PASSWORD=`random_string` && print_info "OTRS_DB_PASSWORD not set, setting password to '${OTRS_DB_PASSWORD}'" && mysqlcmd="mysql -u${OTRS_DB_USER} -h ${OTRS_DB_HOST} -p${OTRS_DB_PASSWORD} "
```
Comment (juanluisbaptiste) [inline otrs/functions.sh]: Why this ?? why  redefine the mysqlcmd variable here ? please remove this.

File: otrs/functions.sh (hunk 3 above)
Comment (wodka) [reply otrs/functions.sh]: yes, fixed

**Comment by wodka** [comment]
updated everything

**Comment by juanluisbaptiste** [comment]
It looks good, one last thing: 

Please apply it to otrs-6_0_x branch as that one will become _master_ in a few days after I finish testing of the OTRS 6 image. Then I can backport this to otrs-5_0_x branch and release a new 5.0.x image with this change when the next OTRS 5.x is released.

**Comment by wodka** [comment]
already did that :) - target is otrs-6_0_x

**Comment by juanluisbaptiste** [comment]
Ahh yes, I didn't notice that, I'll merge then.

Thanks.

**Comment by juanluisbaptiste** [comment]
Did you test your code ? there were a couple errors that prevented the container from correctly starting, see the latest commits.

**Comment by juanluisbaptiste** [comment]
This commit: https://github.com/juanluisbaptiste/docker-otrs/commit/f225280d8663c7335ad83958b31407b63150c62f

**Comment by juanluisbaptiste** [comment]
Also take a look at https://github.com/juanluisbaptiste/docker-otrs/commit/8953165e4cb706329475605365d0b6bfb6c3b215 for a new wait_for_db implementation that doesn't query a table but uses mysql ping instead.
